### DISCUSSION TITLE
Feature Request: Port mod_auth_openidc to Support IBM HTTPD on IBMi Platform

### DISCUSSION THREAD

**Discussion by fcristini** [OpenIDC/mod_auth_openidc#1198] [Ideas]
I would like to request the porting of mod_auth_openidc to support IBM HTTP Server (IBM HTTPD) on the IBMi platform. Currently, mod_auth_openidc is not available for IBMi, and having support for it would greatly enhance the authentication capabilities for IBMi users.
IBMi HTTPD currently provides a number of Apache Authentication modules but no one for OAuth2 ([https://www.ibm.com/docs/en/ibm-http-server/9.0.5?topic=server-apache-modules-included-http](https://www.ibm.com/docs/en/ibm-http-server/9.0.5?topic=server-apache-modules-included-http))

### Motivation:

The porting of mod_auth_openidc to IBM HTTPD on IBMi would offer several significant benefits:

- **OAuth2 Authentication:** This feature would enable IBMi users to utilize OAuth2 for authentication, providing a secure and standardized method for accessing IBMi-based applications.
- **Integration with Keycloak and Other OAuth Providers:** With mod_auth_openidc support, IBMi users could easily integrate their systems with OAuth providers like Keycloak, enabling centralized authentication management and single sign-on capabilities for IBMi applications.

### Additional Information:

- **Current Status:** As of now, mod_auth_openidc does not support IBM HTTP Server on IBMi. This feature request aims to fill this gap and enhance the authentication capabilities for IBMi users.
- **Impact:** The availability of mod_auth_openidc for IBMi would benefit a wide range of users and organizations relying on IBM HTTP Server for hosting their applications, providing them with modern authentication capabilities.

**Comment by zandbelt** [comment]
the IBM HTTP server builds are provided under a commercial license as mentioned in the release notes https://github.com/OpenIDC/mod_auth_openidc/releases/

**Reply by fcristini** [reply]
thank you very much, I'll write to sales@openidc.com for more informations
